dali/operators/image/color/color_twist.h

```diff
@@ -12,252 +12,258 @@
+          , output_type_arg_(spec.GetArgument<DALIDataType>(color::kOutputType))
+          , output_type_(DALI_NO_TYPE) {
+    if (std::is_same<Backend, GPUBackend>::value) {
+      kernel_manager_.Resize(1, 1);
+    } else {
+      kernel_manager_.Resize(num_threads_, batch_size_);
     }
   }
 
-  void Prepare(Index i, const OpSpec &spec, const ArgumentWorkspace * ws) override {
-    contrast_ = spec.GetArgument<float>("contrast", ws, i);
-  }
 
- private:
-  float contrast_;
-};
+  bool CanInferOutputs() const override {
+    return true;
+  }
 
-class Hue : public ColorAugment {
- public:
-  void operator() (float * matrix) override {
-    float temp[nDim*nDim];  // NOLINT(*)
-    for (int i = 0; i < nDim * nDim; ++i) {
-        temp[i] = matrix[i];
-    }
-    const float U = cos(hue_ * M_PI / 180.0);
-    const float V = sin(hue_ * M_PI / 180.0);
-
-    // Single matrix transform for both hue and saturation change. Matrix taken
-    // from https://beesbuzz.biz/code/hsv_color_transforms.php. Derived by
-    // transforming first to HSV, then do the modification, and transfom back to RGB.
-
-    const float const_mat[] = {.299, .587, .114, 0.0,
-                               .299, .587, .114, 0.0,
-                               .299, .587, .114, 0.0,
-                               .0,   .0,   .0,   1.0};
-
-    const float U_mat[] = { .701, -.587, -.114, 0.0,
-                           -.299,  .413, -.114, 0.0,
-                           -.300, -.588, .886,  0.0,
-                            .0,    .0,   .0,    0.0};
-
-    const float V_mat[] = { .168,   .330, -.497, 0.0,
-                           -.328,   .035,  .292, 0.0,
-                           1.25, -1.05,  -.203, 0.0,
-                            .0,    .0,    .0,   0.0};
-    // The last row stays the same so we update only nDim - 1 rows
-    for (int i = 0 ; i < nDim - 1; ++i) {
-      for (int j = 0; j < nDim; ++j) {
-        float sum = 0;
-        for (int k = 0; k < nDim; ++k) {
-          sum += temp[k * nDim + j] * (const_mat[i * nDim + k] +
-                                       U_mat[i * nDim + k] * U +
-                                       V_mat[i * nDim + k] * V);
-        }
-        matrix[i * nDim + j] = sum;
-      }
+  void AcquireArguments(const workspace_t<Backend> &ws) {
+    if (this->spec_.ArgumentDefined(color::kHue)) {
+      this->GetPerSampleArgument(hue_, color::kHue, ws);
+    } else {
+      hue_ = std::vector<float>(batch_size_, color::kHueDefault);
     }
-  }
 
-  void Prepare(Index i, const OpSpec &spec, const ArgumentWorkspace * ws) override {
-    hue_ = spec.GetArgument<float>("hue", ws, i);
-  }
+    if (this->spec_.ArgumentDefined(color::kSaturation)) {
+      this->GetPerSampleArgument(saturation_, color::kSaturation, ws);
+    } else {
+      saturation_ = std::vector<float>(batch_size_, color::kSaturationDefault);
+    }
 
- private:
-  float hue_;
-};
+    if (this->spec_.ArgumentDefined(color::kValue)) {
```

Comment:
I would put some comment here why it is named as `color::kValue` as well.

Reply:
...and I would just use a literal - it's less obscure (and makes is less tempting to change the value in the future ;))